tests/ctl/core/test_api.py

```diff
@@ -1151,6 +1368,119 @@ def test_system_update_privacy_declaration_invalid_duplicate(
             and system.privacy_declarations[1].name == "new declaration 1"
         )
 
+    def test_system_update_dictionary_fields(
+        self,
+        test_config,
+        system_update_request_body_with_new_dictionary_fields,
+        system,
+        db,
+        generate_system_manager_header,
+    ):
+        assert system.name != self.updated_system_name
+
+        auth_header = generate_system_manager_header([system.id])
+        result = _api.update(
+            url=test_config.cli.server_url,
+            headers=auth_header,
+            resource_type="system",
+            json_resource=system_update_request_body_with_new_dictionary_fields.json(
+                exclude_none=True
+            ),
+        )
+
+        json_results = result.json()
+        assert result.status_code == HTTP_200_OK
+        assert json_results["vendor_id"] == "test_vendor"
```

Comment:
same as above, it would be nice to see a more programmatic approach

Reply:
Stepping back for a moment, my preference is for tests to be on the verbose side. I've seen too many programmatic/clever tests that miss when the code changes out from under it. I also think it's valuable to work through each field and verify the results are what you expect.  Of course, there's room for compromise here! I agree with you that here that my tests should be consolidated in part.